staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -319,31 +351,43 @@ func prefixTransformersAndProbes(config apiserverconfig.ResourceConfiguration, s
 			transformer, transformerErr = secretboxPrefixTransformer(provider.Secretbox)
 
 		case provider.KMS != nil:
-			transformer, probe, used, transformerErr = kmsPrefixTransformer(provider.KMS, stopCh)
+			transformer, probe, kmsv2Used, kmsv1Used, transformerErr = kmsPrefixTransformer(provider.KMS, stopCh)
 			if transformerErr == nil {
 				probes = append(probes, probe)
-				kmsUsed.v1Used = kmsUsed.v1Used || used.v1Used
-				kmsUsed.v2Used = kmsUsed.v2Used || used.v2Used
+				kmsv2 = kmsv2 || kmsv2Used
+				kmsv1 = kmsv1 || kmsv1Used
 			}
 
+			// request might take max timeout to work through all the KMS plugins
+			// sum the timeouts for all KMS plugins to ensure that we are allowing sufficient time before closing old transformers
+			kmsMaxTimeoutForResource += provider.KMS.Timeout.Duration
```

Comment:
Shouldn't the max timeout be sufficient to accommodate timeout for plugins? Why are we adding here?

Reply:
https://github.com/kubernetes/kubernetes/pull/112050#discussion_r1014649817

Reply:
updated as per - https://github.com/kubernetes/kubernetes/pull/112050#discussion_r1016056598